### PR TITLE
Fix the double `self`s in module import suggestions

### PR DESCRIPTION
**Pull request by ticki**

### CODE REVIEW

**Comment by arielb1** [comment]
test please

**Comment by Ticki** [comment]
@arielb1 I'm not sure if the error messages are testable. If so, how?

File: src/librustc_resolve/lib.rs [hunk 1]
```diff
@@ -1357,7 +1357,7 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                 let target_mod_str = module_to_string(&*module);
                                 let current_mod_str = module_to_string(&*self.current_module);
 
-                                let prefix = if target_mod_str == current_mod_str {
+                                let prefix = if target_mod_str == current_mod_str && !path_str.starts_with("self::") {
```
Comment (frewsxcv) [inline src/librustc_resolve/lib.rs]: This is longer than 100 characters and needs to be wrapped

**Comment by arielb1** [comment]
`//~ ERROR`

**Comment by alexcrichton** [comment]
Closing due to inactivity, but feel free to resubmit with a test and with the tidy error fixed!

**Pull request closed without merging**